src/metrics/moves.ts

```diff
@@ -1,164 +1,101 @@
-import { Chess } from '../../cjsmin/src/chess';
-import { FileReaderGame } from '../types';
-export async function getGameWithMostMoves(games: FileReaderGame[]) {
-  console.time('Task 5: getGameWithMostMoves');
-  let maxNumMoves = 0;
-  let gameWithMostMoves: FileReaderGame | null = null;
-  let gameLinkWithMostMoves = null;
-
-  for await (const game of games) {
-    const chess = new Chess();
-
-    chess.loadPgn(game.moves);
-    const numMoves = chess.history().length;
-
-    if (numMoves > maxNumMoves) {
-      maxNumMoves = numMoves;
-      gameWithMostMoves = game;
-      let site = game.metadata
-        .find((item) => item.startsWith('[Site "'))
-        ?.replace('[Site "', '')
-        .replace('"]', '');
-      gameLinkWithMostMoves = site;
-    }
+import { Piece, PrettyMove, UASymbol } from '../../cjsmin/src/chess';
+import { UAPMap } from '../types';
+import { createUAPMap } from '../utils';
+import { Metric } from './metric';
+
+export class GameWithMostMovesMetric implements Metric {
+  link: string;
+  numMoves: number;
+
+  constructor() {
+    this.clear();
   }
 
-  console.timeEnd('Task 5: getGameWithMostMoves');
+  clear(): void {
+    this.link = undefined;
+    this.numMoves = 0;
+  }
 
-  // moves facts
-  console.log('MOVES FACTS:');
-  console.log(`The game with the most moves played: ${gameWithMostMoves}`);
-  console.log(`The number of moves played in that game: ${maxNumMoves}`);
+  processGame(game: { move: PrettyMove; board: Piece[] }[], gameLink?: string) {
+    if (game.length > this.numMoves) {
```

Comment:
does game.length return the appropriate value now? i had trouble with this in the past. wouldn't it just be 3 given there are three sub arrays in the game array?

Reply:
should be? Each entry is a move

Reply:
look @ the typing